src/components/InputForm.tsx

```diff
@@ -0,0 +1,15 @@
+import { TextField, TextFieldProps } from '@mui/material';
+import React from 'react';
+import { UseFormRegister } from 'react-hook-form';
+import type { AuthData } from '../types/auth.types';
+
+type InputProps = TextFieldProps & {
+  register?: UseFormRegister<AuthData>;
+};
+
+const InputForm = React.forwardRef<HTMLInputElement, InputProps>((props: InputProps, ref) => {
+  return <TextField ref={ref} fullWidth {...props} />;
```

Comment:
why do you need fullWidth explicitly? doesn't it come from props?

Reply:
no it is not passing as props.
should i pass fullwidth as props?